python/cuml/dask/manifold/umap.py

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import numpy as np
+
+from cuml.dask.common.input_utils import DistributedDataHandler
+from cuml.dask.common.base import DelayedTransformMixin
+from cuml.dask.manifold.base import BaseManifold
+
+from cuml.manifold.umap import UMAP as cumlUMAP
+
+
+class UMAP(BaseManifold,
+           DelayedTransformMixin):
+    def __init__(self, client=None, verbose=False, n_sampling=500, **kwargs):
+        super(UMAP, self).__init__(client, **kwargs)
+        self.local_model = cumlUMAP(verbose=verbose, MNMG=True, **kwargs)
+        self.n_sampling = n_sampling
+
+    def fit(self, X, y=None, convert_dtype=True):
```

Comment:
We might need to think through this one a little bit. Perhaps we could even forego the sampling for the first iteration and just require the user to create a cuml.dask.manifold.UMAP model with an already-trained UMAP model? 

IMO, if we are going to add the fit function to the mix, we probably want to make it consistent with the other distributed models and accept dask.array or dask.dataframe as input, which would then mean sampling from the distributed object for training the model on a single GPU.